currency_rate_update_nbu/models/res_currency_rate_provider_nbu.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2022 Garazd Creation (https://garazd.biz)
+# @author: Yurii Razumovskyi (garazdcreation@gmail.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+import json
+import logging
+from collections import defaultdict
+from typing import List
+
+import dateutil.parser
+import requests
+from requests.exceptions import Timeout, TooManyRedirects
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.float_utils import float_compare
+
+_logger = logging.getLogger(__name__)
+
+
+class ResCurrencyRateProviderNBU(models.Model):
+    """Implementation for National Bank of Ukraine"""
+
+    _inherit = "res.currency.rate.provider"
+
+    service = fields.Selection(
+        selection_add=[("NBU", "National Bank of Ukraine")],
+        ondelete={"NBU": "set default"},
+    )
+
+    def _get_supported_currencies(self):
+        self.ensure_one()
+        if self.service != "NBU":
+            return super()._get_supported_currencies()
+        # List of currencies obrained from:
+        # https://bank.gov.ua/NBUStatService/v1/statdirectory/exchange?json
+        return [
```

Comment:
do not use hardcoded list, do a dynamic call to nbu service and parse json properly